x-pack/plugins/transform/public/app/hooks/__mocks__/use_api.ts

```diff
@@ -4,53 +4,113 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { TransformId, TransformEndpointRequest } from '../../../../common';
+import { HttpFetchError } from 'kibana/public';
 
-import { PreviewRequestBody } from '../../common';
+import { TransformId } from '../../../../common/types/transform';
+import type { GetTransformsAuditMessagesResponseSchema } from '../../../../common/api_schemas/audit_messages';
+import type {
+  DeleteTransformsRequestSchema,
+  DeleteTransformsResponseSchema,
+} from '../../../../common/api_schemas/delete_transforms';
+import type {
+  StartTransformsRequestSchema,
+  StartTransformsResponseSchema,
+} from '../../../../common/api_schemas/start_transforms';
+import type {
+  StopTransformsRequestSchema,
+  StopTransformsResponseSchema,
+} from '../../../../common/api_schemas/stop_transforms';
+import type {
+  GetTransformsResponseSchema,
+  PostTransformsPreviewRequestSchema,
+  PutTransformsRequestSchema,
+  PutTransformsResponseSchema,
+} from '../../../../common/api_schemas/transforms';
+import type { GetTransformsStatsResponseSchema } from '../../../../common/api_schemas/transforms_stats';
+import type {
+  PostTransformsUpdateRequestSchema,
+  PostTransformsUpdateResponseSchema,
+} from '../../../../common/api_schemas/update_transforms';
 
 const apiFactory = () => ({
-  getTransforms(transformId?: TransformId): Promise<any> {
+  getTransform(transformId: TransformId): Promise<GetTransformsResponseSchema | HttpFetchError> {
     return new Promise((resolve, reject) => {
-      resolve([]);
+      resolve({ count: 0, transforms: [] });
```

Comment:
nit, these could be changed to 
`return Promise.resolve({ count: 0, transforms: [] })`

Reply:
Done in a35ea1e.